azuread/resource_group_test.go

```diff
@@ -116,10 +231,112 @@ func testCheckAzureADGroupDestroy(s *terraform.State) error {
 	return nil
 }
 
+func assertResourceWithMemberCount(id string, memberCount string) resource.TestCheckFunc {
+	resourceName := "azuread_group.test"
+
+	return resource.ComposeTestCheckFunc(
+		testCheckAzureADGroupExists(resourceName),
+		resource.TestCheckResourceAttr(resourceName, "name", fmt.Sprintf("acctest%s", id)),
+		resource.TestCheckResourceAttrSet(resourceName, "object_id"),
+		resource.TestCheckResourceAttr(resourceName, "members.#", memberCount),
+	)
+}
+
 func testAccAzureADGroup(id string) string {
 	return fmt.Sprintf(`
 resource "azuread_group" "test" {
   name = "acctest%s"
 }
 `, id)
 }
+
+func testAccAzureADGroupWithDiverseMembers(id string) string {
+	var sb strings.Builder
```

Comment:
Could we just use a large here string here with a fmt.Sprintf for consistency with the rest of the provider?

Reply:
Not using `fmt.Sprintf` gives us a few benefits:

- Reuse
- Looping and other code functionality

On the other hand I can understand an argument of making it less obvious what it does since the HCL is not visible in plain text.

Reply:
I prefer the current approach but if it's not feasible I could try to rewrite the test configuration.

Reply:
We definitely would prefer HCL as this raises the barriers for maintainers & community contributors as well as people will often copy and past the tests for their use. And most importantly is not consistent with any of the azure provider tests.

if you'd like to loop maybe just have a simple loop at the start appending to a string.